erpnext/patches/v13_0/update_customer_total_sales.py

```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2017, Frappe and Contributors
+# License: GNU General Public License v3. See license.txt
+
+from __future__ import unicode_literals
+import frappe
+from erpnext.selling.doctype.customer.customer import update_customer_current_month_sales
+
+def execute():
+	'''Update customer monthly sales history based on sales invoices'''
+	frappe.reload_doctype("Customer")
+	customers = [d['name'] for d in frappe.get_list("Customer")]
+
+	for customer in customers:
+		update_customer_current_month_sales(customer)
```

Comment:
This patch will take lot of time if the number of customers are more than 1,00,000

Reply:
@rohitwaghchaure  Can you please suggest something for this?

Reply:
get customer wise sales data from sales invoice using sql query(group by customer) and then update the value in the customer. If particular instance has 10,000 customer out of which only 100 customers has the sales invoices and not all then no need to check for each customer.